In [1]:
import pandas as pd
import numpy as np

In [2]:
Sales_data=pd.read_excel('Sales.xlsx')

In [3]:
Old=Sales_data['Old Scheme (in thousands)']
New=Sales_data['New Scheme (in thousands)']

In [4]:
np.mean(Old)

68.03333333333333

In [5]:
np.mean(New)

72.03333333333333

# Performing one tailed test

In [6]:
from scipy.stats import ttest_ind,ttest_1samp


In [7]:
ttest_ind(Old,New)

Ttest_indResult(statistic=-0.6937067608923764, pvalue=0.49063515686248105)

#  2)Since p<0.5 we failed to reject null hypothesis i,e  new scheme has significantly NOT raised output

# 3)Probability of Type I error = significant level = 0.05 or 5%

In [8]:
t,p=ttest_1samp(New-Old,5)
print(p)


0.7001334912613286


# Since it is a one sided test Pvalue 0.700 which is greater than level of significance α (0.05) , hence we retain the null hypothesis or we fail to reject the Null Hypothesis on

# Power of test

In [9]:
from statsmodels.stats.power import ttest_power

In [10]:
(np.mean(Old) - np.mean(New)) / np.sqrt(((30-1)*np.var(Old)+(30-1)*np.var(New)) / 60-2)

-0.1856895502907714

In [11]:
stdp=np.sqrt((np.std(Old)**2+np.std(New)**2)/2)

In [13]:
(np.mean(Old)-np.mean(New))/stdp

-0.18217632044690082

In [14]:
ttest_power(0.182,nobs=60,alpha=0.10)

0.40199207962675076

# Power of test : 0.40199207962675076